apps/server/src/modules/group/saga/delete-user-group-data.step.ts

```diff
@@ -0,0 +1,78 @@
+import { Logger } from '@core/logger';
+import {
+	ModuleName,
+	SagaService,
+	SagaStep,
+	StepOperationReport,
+	StepOperationReportBuilder,
+	StepOperationType,
+	StepReport,
+	StepReportBuilder,
+	StepStatus,
+	UserDeletionStepOperationLoggable,
+} from '@modules/saga';
+import { Injectable, InternalServerErrorException } from '@nestjs/common';
+import { EntityId } from '@shared/domain/types';
+import { Group } from '../domain';
+import { GroupRepo } from '../repo';
```

Comment:
The step could be seen more like a use case. So it shouldn't maybe use the `GroupRepo` directly but a public domain service that internally calls the repo.